### PR TITLE
Only deploy collectors for supported applications

### PR DESCRIPTION
**Pull request by agrare**
The only applications that depend on topological-inventory data and thus
require collectors to be deployed are catalog and topological-inventory.

If a source isn't enabled for any of these application_types then we
shouldn't deploy collectors for it.

Fixes https://projects.engineering.redhat.com/browse/TPINVTRY-546

### CODE REVIEW

**Comment by agrare** [comment]
We could probably make this more efficient by doing one request to `/applications` per-tenant for the supported application_type_ids but wanted to get started on this.

We can always make performance enhancements as a follow-up

File: lib/topological_inventory/orchestrator/worker.rb [hunk 1]
```diff
@@ -70,6 +71,11 @@ def each_source
 
             next unless (collector_definition = collector_definitions[source_type["name"]])
 
+            application_types = get_and_parse(sources_api_url_for("sources", source["id"], "application_types"), tenant)
+
+            enabled_applications = application_types&.dig("data")&.map { |app_type| app_type["name"] } || []
```
Comment (bdunne) [inline lib/topological_inventory/orchestrator/worker.rb]: Will `application_types` ever be `nil`?

Comment (agrare) [reply lib/topological_inventory/orchestrator/worker.rb]: Ideally that's correct, `get_and_parse` does return `nil` on a `RestClient::NotFound` exception though not sure if that's something we need to handle here

Comment (agrare) [reply lib/topological_inventory/orchestrator/worker.rb]: I could just `next if application_types.nil?` and avoid the `&. &.`, if it is non-nil then it should have the proper `"data"` key, wyt?

Comment (bdunne) [reply lib/topological_inventory/orchestrator/worker.rb]: I think the only time we would get a `RestClient::NotFound` is if the source doesn't exist and [we already account for that](https://github.com/ManageIQ/topological_inventory-orchestrator/pull/37/files#diff-4743ee12ee7e621468f2e6590de994efR68)

Comment (agrare) [reply lib/topological_inventory/orchestrator/worker.rb]: :+1: done

**Pull request merged**